votor/src/voting_utils.rs

```diff
@@ -313,3 +311,293 @@ pub fn generate_vote_message(
+        let bank_forks = BankForks::new_rw_arc(bank0);
+
+        let my_keys = &validator_keypairs[my_index];
+        let root_bank = bank_forks.read().unwrap().sharable_root_bank();
+        VotingContext {
+            vote_history: VoteHistory::new(my_keys.node_keypair.pubkey(), 0),
+            vote_account_pubkey: my_keys.vote_keypair.pubkey(),
+            identity_keypair: Arc::new(my_keys.node_keypair.insecure_clone()),
+            authorized_voter_keypairs: Arc::new(RwLock::new(vec![Arc::new(
+                my_keys.vote_keypair.insecure_clone(),
+            )])),
+            derived_bls_keypairs: HashMap::new(),
+            has_new_vote_been_rooted: false,
+            own_vote_sender,
+            bls_sender: unbounded().0,
+            commitment_sender: unbounded().0,
+            wait_to_vote_slot: None,
+            root_bank,
+        }
+    }
+
+    #[test]
+    fn test_generate_vote_message() {
+        let (own_vote_sender, own_vote_receiver) = crossbeam_channel::unbounded();
+        // Create 10 node validatorvotekeypairs vec
+        let validator_keypairs = (0..10)
+            .map(|_| ValidatorVoteKeypairs::new(Keypair::new(), Keypair::new(), Keypair::new()))
+            .collect::<Vec<_>>();
+        let my_index = 0;
+        let mut voting_context =
+            setup_voting_context_and_bank_forks(own_vote_sender, &validator_keypairs, my_index);
+        let my_bls_keypair = BLSKeypair::derive_from_signer(
+            &validator_keypairs[my_index].vote_keypair,
+            BLS_KEYPAIR_DERIVE_SEED,
+        )
+        .unwrap();
+
+        // Generate a normal notarization vote and check it's sent out correctly.
+        let block_id = Hash::new_unique();
+        let vote_slot = 2;
+        let vote = Vote::new_notarization_vote(vote_slot, block_id);
+        let result = generate_vote_message(vote, false, &mut voting_context)
+            .ok()
+            .unwrap()
+            .unwrap();
+        let expected_message = generate_expected_consensus_message(vote, &my_bls_keypair);
+        if let BLSOp::PushVote {
+            message,
+            slot,
+            saved_vote_history,
+        } = &result
+        {
+            assert_eq!(slot, &vote_slot);
+            assert_eq!(**message, expected_message);
+            assert_eq!(
+                saved_vote_history,
+                &SavedVoteHistoryVersions::from(
+                    SavedVoteHistory::new(
+                        &voting_context.vote_history,
+                        &voting_context.identity_keypair
+                    )
+                    .unwrap()
+                )
+            );
+        } else {
+            panic!("Expected BLSOp::PushVote, got {result:?}");
+        }
+
+        // Check that own vote sender receives the vote
+        let received_message = own_vote_receiver.recv().unwrap();
+        assert_eq!(received_message, expected_message);
+    }
+
+    #[test]
+    fn test_wrong_vote_account() {
```

Comment:
nit: this covers more than just the wrong vote account. Maybe  break the other edge cases into separate tests